File: backend/Bingol-University-Social-Media-main/src/main/java/bingol/campus/friendRequest/core/response/FollowedUserDTO.java
package bingol.campus.friendRequest.core.response;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class FollowedUserDTO {
    private long id;
    private String username;           // Kullanıcı adı
    private String fullName;           // Tam ad
    private String profilePhotoUrl;    // Profil fotoğrafı URL'si
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private LocalDate followedDate; // Takip edildiği tarih
    private boolean isActive;          // Takip ilişkisi aktif mi?
    private boolean isPrivate;
    private String bio;                // Kullanıcının biyografisi
    private int popularityScore;       // Kullanıcının popülerlik skoru
}

File: frontend/lib/models/followed_user_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'followed_user_dto.g.dart';

@JsonSerializable()
class FollowedUserDTO {
  final int id; // long türü Dart'ta int olarak temsil edilir
  final String username; // Kullanıcı adı
  final String fullName; // Tam ad
  final String? profilePhotoUrl; // Profil fotoğrafı URL'si

  @JsonKey(name: 'followedDate')
  final DateTime followedDate; // Takip edildiği tarih

  @JsonKey(defaultValue: false)
  final bool isActive; // Takip ilişkisi aktif mi?
  
  @JsonKey(defaultValue: false)
  final bool isPrivate; // Kullanıcı özel mi?
  
  final String? bio; // Kullanıcının biyografisi
  final int popularityScore; // Kullanıcının popülerlik skoru
  
  @JsonKey(name: 'isFollowing', defaultValue: false)
  final bool isFollowing; // Kullanıcı takip ediliyor mu?

  FollowedUserDTO({
    required this.id,
    required this.username,
    required this.fullName,
    this.profilePhotoUrl,
    required this.followedDate,
    required this.isActive,
    required this.isPrivate,
    this.bio,
    required this.popularityScore,
    required this.isFollowing,
  });

  factory FollowedUserDTO.fromJson(Map<String, dynamic> json) => _$FollowedUserDTOFromJson(json);
  Map<String, dynamic> toJson() => _$FollowedUserDTOToJson(this);
} 